api/src/integrations/nris/datasource.js

```diff
@@ -0,0 +1,236 @@
+const axios               = require('axios');
+
+const NRIS_TOKEN_ENDPOINT   = process.env.NRIS_TOKEN_ENDPOINT || "https://api.nrs.gov.bc.ca/oauth2/v1/oauth/token?disableDeveloperFilter=true&grant_type=client_credentials&scope=NRISWS.*";
+const NRIS_EPD_API_ENDPOINT = process.env.NRIS_EPD_API_ENDPOINT || "https://api.nrs.gov.bc.ca/nrisws-api/v1/epdInspections";
+const NRIS_username         = process.env.NRIS_username || null;
+const NRIS_password         = process.env.NRIS_password || null;
+
+class NrisDataSource {
+  /**
+   * Creates an instance of NrisDataSource.
+   *
+   * @param {*} auth_payload information about the user account that started this update.
+   * @param {*} [params=null] params to filter records (optional).
+   * @param {*} [recordTypes=null] specific record types to update (optional).
+   * @memberof NrisDataSource
+   */
+  constructor(auth_payload, params = null, recordTypes = null) {
+    this.auth_payload = auth_payload;
+    this.params = params || {};
+
+    // Set initial status
+    this.status = { itemsProcessed: 0, itemTotal: 0, typeStatus: [] };
+  }
+
+  // Setup the datasource and start running the task.
+  async run(taskAuditRecord) {
+    await taskAuditRecord.updateTaskRecord({ status: 'Running' });
+
+    // First perform authentication against this datasource
+    // We should login using env var creds - get a token from our configured endpoint.
+    this.token = null;
+
+    if (NRIS_username === null || NRIS_password === null) {
+      defaultLog.error('Must set environment username/password for NRIS data connection.')
+      return { status: 'Configuration Error' };
+    }
+
+    try {
+      const res = await axios.get(NRIS_TOKEN_ENDPOINT,
+                                  {
+                                    timeout: 2000,
+                                    auth: {
+                                      username: NRIS_username,
+                                      password: NRIS_password
+                                    }
+                                  });
+
+      const payload = res.data ? res.data : null;
+
+      if (!payload) {
+        defaultLog.error('Error authenticating against NRIS API.');
+        return { status: 'Auth Error' };
+      }
+
+      this.token = payload.access_token;
+      defaultLog.info('NRIS API token expires:', (payload.expires_in / 60 / 60).toFixed(2), ' hours');
+  
+      // Hardcoded to start in 2017
+      let startDate = moment('2017-01-01');
+      let endDate = moment(startDate).add(1, 'M');
+
+      let statusObject = {
+        status: 'Complete',
+        message: 'Job Complete',
+        itemsProcessed: 0,
+        itemTotal: 0,
+      };
+
+      // Keep going until we'd start past today's date.
+      while(startDate < moment()) {
+        defaultLog.info("dateRange:", startDate.format('YYYY-MM-DD'), endDate.format('YYYY-MM-DD'));
+        startDate = endDate;
+        endDate = moment(startDate).add(1, 'M');
```

Comment:
These dates should increment at the end of the while loop.
Currently, the first month of 2017 is being missed.